// @wordpress/theme/build-module/theme-provider.mjs
// packages/theme/src/theme-provider.tsx
import { useMemo, useId } from "@wordpress/element";
import { ThemeContext } from "./context.mjs";
import { useThemeProviderStyles } from "./use-theme-provider-styles.mjs";

// packages/style-runtime/src/index.ts
var STYLE_HASH_ATTRIBUTE = "data-wp-hash";
function getRuntime() {
  const globalScope = globalThis;
  if (globalScope.__wpStyleRuntime) {
    return globalScope.__wpStyleRuntime;
  }
  globalScope.__wpStyleRuntime = {
    documents: /* @__PURE__ */ new Map(),
    styles: /* @__PURE__ */ new Map(),
    injectedStyles: /* @__PURE__ */ new WeakMap()
  };
  if (typeof document !== "undefined") {
    registerDocument(document);
  }
  return globalScope.__wpStyleRuntime;
}
function documentContainsStyleHash(targetDocument, hash) {
  if (!targetDocument.head) {
    return false;
  }
  for (const style of targetDocument.head.querySelectorAll(
    `style[${STYLE_HASH_ATTRIBUTE}]`
  )) {
    if (style.getAttribute(STYLE_HASH_ATTRIBUTE) === hash) {
      return true;
    }
  }
  return false;
}
function injectStyle(targetDocument, hash, css) {
  if (!targetDocument.head) {
    return;
  }
  const runtime = getRuntime();
  let injectedStyles = runtime.injectedStyles.get(targetDocument);
  if (!injectedStyles) {
    injectedStyles = /* @__PURE__ */ new Set();
    runtime.injectedStyles.set(targetDocument, injectedStyles);
  }
  if (injectedStyles.has(hash)) {
    return;
  }
  if (documentContainsStyleHash(targetDocument, hash)) {
    injectedStyles.add(hash);
    return;
  }
  const style = targetDocument.createElement("style");
  style.setAttribute(STYLE_HASH_ATTRIBUTE, hash);
  style.appendChild(targetDocument.createTextNode(css));
  targetDocument.head.appendChild(style);
  injectedStyles.add(hash);
}
function registerDocument(targetDocument) {
  const runtime = getRuntime();
  runtime.documents.set(
    targetDocument,
    (runtime.documents.get(targetDocument) ?? 0) + 1
  );
  for (const [hash, css] of runtime.styles) {
    injectStyle(targetDocument, hash, css);
  }
  return () => {
    const count = runtime.documents.get(targetDocument);
    if (count === void 0) {
      return;
    }
    if (count <= 1) {
      runtime.documents.delete(targetDocument);
      return;
    }
    runtime.documents.set(targetDocument, count - 1);
  };
}
function registerStyle(hash, css) {
  const runtime = getRuntime();
  runtime.styles.set(hash, css);
  for (const targetDocument of runtime.documents.keys()) {
    injectStyle(targetDocument, hash, css);
  }
}

// packages/theme/src/style.module.css
if (typeof process === "undefined" || process.env.NODE_ENV !== "test") {
  registerStyle("f4e6e06c6a", ".dba930ea7a9438fd__root{display:contents}");
}
var style_default = { "root": "dba930ea7a9438fd__root" };

// packages/theme/src/theme-provider.tsx
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
function cssObjectToText(values) {
  return Object.entries(values).map(([key, value]) => `${key}: ${value};`).join("");
}
function generateCSSSelector({
  instanceId,
  isRoot
}) {
  const rootSel = `[data-wpds-root-provider="true"]`;
  const instanceIdSel = `[data-wpds-theme-provider-id="${instanceId}"]`;
  const selectors = [];
  if (isRoot) {
    selectors.push(
      `:root:has(.${style_default.root}${rootSel}${instanceIdSel})`
    );
  }
  selectors.push(`.${style_default.root}.${style_default.root}${instanceIdSel}`);
  return selectors.join(",");
}
var ThemeProvider = ({
  children,
  color = {},
  cursor,
  isRoot = false,
  density
}) => {
  const instanceId = useId();
  const { themeProviderStyles, resolvedSettings } = useThemeProviderStyles({
    color,
    cursor
  });
  const contextValue = useMemo(
    () => ({
      resolvedSettings
    }),
    [resolvedSettings]
  );
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    themeProviderStyles ? /* @__PURE__ */ jsx("style", { children: `${generateCSSSelector({
      instanceId,
      isRoot
    })} {${cssObjectToText(themeProviderStyles)}}` }) : null,
    /* @__PURE__ */ jsx(
      "div",
      {
        "data-wpds-theme-provider-id": instanceId,
        "data-wpds-root-provider": isRoot,
        "data-wpds-density": density,
        className: style_default.root,
        children: /* @__PURE__ */ jsx(ThemeContext.Provider, { value: contextValue, children })
      }
    )
  ] });
};
export {
  ThemeProvider
};
